draft-tus-resumable-uploads-protocol.md

```diff
@@ -232,6 +230,8 @@ upload-offset: 25
 
 The client MAY automatically attempt upload resumption when the connection is terminated unexpectedly, or if a server error status code between 500 and 599 (inclusive) is received. The client SHOULD NOT automatically retry if a client error status code between 400 and 499 (inclusive) is received.
 
+The client MAY send the metadata of the file using headers such as `Content-Type` (see {{Section 8.3 of HTTP}} and `Content-Disposition` {{!RFC6266}} when starting a new upload. If so, the client MUST only include these header fields in the initial Upload Transfer Procedure, which starts the new upload. The server MUST ignore these header fields in subsequent Upload Transfer Procedures. The `Content-Type` header MAY include the MIME type of the file. The `Content-Disposition` header MAY include the `filename`, `filename*` or `boundary` parameters. The server MAY interpret this data or MAY choose to ignore it. 
```

Comment:
As discussed in the meeting, MUST is too strict. Client might not know which headers are metadata

Reply:
I suggest modifying this paragraph a bit to something like below. The suggestion is based on my understanding so might be incorrect but the structure is the important part. HTTP supports representation meadata so we don't need to say `MAY` on requests - we are not redefining the rules of HTTP AFAICT.

Guoye's comment should be accommodated but I kept it in my suggestion for clarity.


> File metadata can affect how servers might act on the uploaded file. Clients can send Representation Metadata (see {{Section 8.3 of HTTP}}) in the Upload Transfer Procedure request that starts an upload.  Servers MAY interpret this metadata or MAY ignore it. Clients MUST NOT send representation metadata in subsequent Upload Transfer Procedure requests; servers MUST ignore representation received in subsequent requests. Where a client includes `Content-Disposition` the parameters MUST be either `filename`, `filename*` or `boundary`.

Reply:
Thank you for your feedback! Included it in the PR.